paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java

```diff
@@ -94,26 +94,36 @@ public void createTag(
             List<TagCallback> callbacks) {
         checkArgument(!StringUtils.isBlank(tagName), "Tag name '%s' is blank.", tagName);
 
-        // skip create tag for the same snapshot of the same name.
+        // When timeRetained is not defined, please do not write the tagCreatorTime field,
+        // as this will cause older versions (<= 0.7) of readers to be unable to read this
+        // tag.
+        // When timeRetained is defined, it is fine, because timeRetained is the new
+        // feature.
+        String content =
+                timeRetained != null
+                        ? Tag.fromSnapshotAndTagTtl(snapshot, timeRetained, LocalDateTime.now())
+                                .toJson()
+                        : snapshot.toJson();
+
+        // update tag metadata into for the same snapshot of the same tag name.
         if (tagExists(tagName)) {
             Snapshot tagged = taggedSnapshot(tagName);
             Preconditions.checkArgument(
                     tagged.id() == snapshot.id(), "Tag name '%s' already exists.", tagName);
+
+            Path tagPath = tagPath(tagName);
+            try {
+                fileIO.overwriteFileUtf8(tagPath, content);
+            } catch (IOException ignored) {
+                LOG.info(
```

Comment:
why catch the exception?

Reply:
method overwriteFileUtf8 maybe throw IOException and createTag does not process it. we just print the info when overwriteFileUtf8 failed.

Reply:
So why not throw RuntimeException?

Reply:
got. you are right. we should mark it failed.